src/collection.js

```diff
@@ -230,6 +237,52 @@ export default class Collection {
   }
 
   /**
+   * Validate the passed hook is correct.
+   *
+   * @param {Array|undefined} hook.
+   * @return {Array}
+   **/
+  _validateHook(hook) {
+    if (!Array.isArray(hook)) {
+      throw new Error("A hook definition should be an array of functions.");
+    }
+    return hook.map(fun => {
+      if (typeof fun !== "function") {
+        throw new Error("A hook definition should be an array of functions.");
+      }
+      return fun;
+    });
+  }
+
+  /**
+   * Validates a list of hooks.
+   *
+   * @param  {Object|undefined} hooks
+   * @return {Object}
+   */
+  _validateHooks(hooks) {
+    if (typeof hooks === "undefined") {
+      return {};
+    }
+    if (Array.isArray(hooks)) {
+      throw new Error("hooks should be an object, not an array.");
```

Comment:
Sorry, I'm missing previous context it seems. If you want to pass the collection instance as a second arg to avoid creating factories, that's a very good reason for doing so :)